src/main/java/com/deloitte/ddwatch/exceptions/DetailedException.java

```diff
@@ -0,0 +1,16 @@
+package com.deloitte.ddwatch.exceptions;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.http.HttpStatus;
+
+@Data @NoArgsConstructor @AllArgsConstructor
+public abstract class DetailedException extends RuntimeException {
+    protected HttpStatus status;
```

Comment:
check to see if @ResponseStatus meets or requirements

Reply:
Kept the ControllerAdvice for exceptions which have messages, with different handlers per class. Replaced DetailedException with the Spring provided ReponseStatusException.